### PR TITLE
DLS-3909 DisposalDate and ExchangeDate added

### PR DESCRIPTION
**Pull request by rajanimohan22**
DisposalDate and ExchangeDate added

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/SingleDisposalsTriageController.scala [hunk 1]
```diff
@@ -450,7 +450,12 @@ class SingleDisposalsTriageController @Inject() (
           displayTriagePage(state, triageAnswers)(
             _.fold(_.assetType, c => Some(c.assetType)),
             answers => disposalDateBackLink(answers)
-          )(_ => disposalDateForm(TimeUtils.today(), personalRepDetails))(
+          )(_ =>
+            disposalDateForm(
+              TimeUtils.getMaximumDateOf(TimeUtils.today(), TimeUtils.maxDateForSinglePropertyDisposals),
```
Comment (rambabu-posa) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/SingleDisposalsTriageController.scala:455]: We need to build this logic with future flagged. Its applicable to all places

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/TimeUtils.scala [hunk 2]
```diff
@@ -40,6 +40,8 @@ object TimeUtils {
 
   val minimumDate: LocalDate = LocalDate.of(1900, 1, 1)
 
+  val maxDateForSinglePropertyDisposals: LocalDate = LocalDate.of(2022, 4, 6)
```
Comment (rambabu-posa) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/TimeUtils.scala:43]: It should work for future tax years too, no hardcoded way

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/SingleDisposalsTriageControllerSpec.scala [hunk 3]
```diff
@@ -2097,7 +2097,7 @@ class SingleDisposalsTriageControllerSpec
           gainOrLossAfterReliefs = None
         )
 
-      val tomorrow = today.plusDays(1L)
+      val tomorrow = TimeUtils.getMaximumDateOf(today, TimeUtils.maxDateForSinglePropertyDisposals).plusDays(1L)
```
Comment (rambabu-posa) [inline test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/SingleDisposalsTriageControllerSpec.scala:2100]: try to unit test future flag here

**Comment by rambabu-posa** [comment]
LGTM

**Review by rambabu-posa — COMMENTED** [review]
please do these changes:
enabled = true  for local
move getMaximumDateForDisposalsAndCompletion from controller to TimeUtils
Single Indirect disposals for disposalDateOfShares() form is taking today's date as max date

**Comment by rambabu-posa** [comment]
```
  def getMaximumDateForDisposalsAndCompletion(
    enableFutureDateForDisposalAndCompletion: Boolean,
    maxYearForDisposalsAndCompletion: Int
  ): LocalDate =
    if (enableFutureDateForDisposalAndCompletion)
      TimeUtils.getMaximumDateOf(TimeUtils.today(), LocalDate.of(maxYearForDisposalsAndCompletion, 4, 5))
    else
      TimeUtils.today()
```

File: conf/application.conf [hunk 4]
```diff
@@ -259,6 +259,15 @@ mongodb.uri = "mongodb://localhost:27017/cgt-property-disposals-frontend"
 
 enable-language-switching = true
 
+enable-futureDate-Completion = true
```
Comment (calinursaciuc) [inline conf/application.conf:262]: I think this is not needed? You have 2 booleans.

**Review by rambabu-posa — APPROVED** [review]
LGTM

**Pull request merged**